### PR TITLE
Cleanup on Offline Membership Receipts (ensures variables present across flows)

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
This

1) adds shell classes for the membership receipts
2) uses a standardised 'receipt_text' across both forms
(this value is entered on the form rather than being from the db
so it is temporaral - which a smarty variable is a good use for)
3) removes isset from the templates - these currently cause
crashes with grumpy smarty mode

Before
----------------------------------------
- `$contributionID` not available from renewal per @mlutfy 
- isset causes grumpy mode to crash

After
----------------------------------------
- consistent assigns & for `$contributionID`, `$membershipID`, `$contactID`.
-  isset is gone

Technical Details
----------------------------------------
This alters new-install templates but there is no auto-update

Comments
----------------------------------------

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
@monishdeb perhaps if @mlutfy doesn't have time to look you could?

File: CRM/Batch/Form/Entry.php [hunk 1]
```diff
@@ -937,9 +937,6 @@ protected function emailReceipt($form, &$formValues): bool {
     $form->assign('module', 'Membership');
     $form->assign('contactID', $formValues['contact_id']);
 
-    $form->assign('membershipID', CRM_Utils_Array::value('membership_id', $form->_params, CRM_Utils_Array::value('membership_id', $form->_defaultValues)));
-    $this->assign('contributionID', $this->getCurrentRowContributionID());
```
Comment (mlutfy) [inline CRM/Batch/Form/Entry.php:941]: Just double-checking: if someone customized their receipt templates to use `membershipID` and `contributionID`, now they should update to use `membershipId` / `contributionId`? If yes, could we support both? (hard to guess smarty variables often make devs cry)

**Comment by mlutfy** [comment]
Based on code review, I learned a few things, and I like the new style.

It's difficult for me to test, because I am hesitant to throw this patch into production. With the test, and considering it's Eileen, I would feel comfortable merging, but another set of eyes might be best.

Just a small question regarding the smarty variable names (see other comment, around the `membershipID` vs `membershipId` assigns).

File: CRM/Batch/Form/Entry.php (hunk 1 above)
Comment (eileenmcnaughton) [reply CRM/Batch/Form/Entry.php:941]: @mlutfy no the smarty variables are `$contributionID` and `$membershipID`

**Comment by eileenmcnaughton** [comment]
@mlutfy so it is assigned to the template as `$membershipID` based on this line https://github.com/civicrm/civicrm-core/pull/22736/files#diff-90e29fbb63a48e63a1ce23c2b83f0d14d0c0f542630b85fa7c04fd513f397815R22 - and if you check the test it checks the proper camel case variables `$contributionID` etc. Unfortunately the token system uses that `contributionId` style - I did try to fix it once but it was potentially a breaking change

**Comment by mlutfy** [comment]
@eileenmcnaughton Thanks for the clarifications!  I re-read the changes and did a quick test, seems good to me.

**Comment by eileenmcnaughton** [comment]
thanks @mlutfy

**Comment by demeritcowboy** [comment]
I've added https://github.com/civicrm/civicrm-core/pull/22821 since otherwise after upgrade I get a warning about outdated tokens in use.

**Comment by eileenmcnaughton** [comment]
thanks @demeritcowboy

**Pull request merged**